// lowcoder-sdk/dist/2812743e.js
import F from "react";
import { c as N, w as z, a as de, m as pe, u as ve, S as Ee } from "./dbce84da.js";
import { j as T, F as H, a as B } from "./8c706935.js";
import "react-dom";
const P = "adm-badge", X = /* @__PURE__ */ T(H, {}), _e = (s) => {
  const {
    content: f,
    color: c,
    children: i
  } = s, l = f === X, d = N(P, !!i && `${P}-fixed`, l && `${P}-dot`, s.bordered && `${P}-bordered`), v = f || f === 0 ? z(s, /* @__PURE__ */ T("div", {
    className: d,
    style: {
      "--color": c
    },
    children: !l && /* @__PURE__ */ T("div", {
      className: `${P}-content`,
      children: f
    })
  })) : null;
  return i ? /* @__PURE__ */ B("div", {
    className: N(`${P}-wrapper`, s.wrapperClassName),
    style: s.wrapperStyle,
    children: [i, v]
  }) : v;
}, G = de(_e, {
  dot: X
});
var Z = { exports: {} }, a = {};
/** @license React v17.0.2
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var U;
function me() {
  if (U)
    return a;
  U = 1;
  var s = 60103, f = 60106, c = 60107, i = 60108, l = 60114, d = 60109, v = 60110, n = 60112, u = 60113, C = 60120, p = 60115, E = 60116, R = 60121, A = 60122, h = 60117, y = 60129, S = 60131;
  if (typeof Symbol == "function" && Symbol.for) {
    var t = Symbol.for;
    s = t("react.element"), f = t("react.portal"), c = t("react.fragment"), i = t("react.strict_mode"), l = t("react.profiler"), d = t("react.provider"), v = t("react.context"), n = t("react.forward_ref"), u = t("react.suspense"), C = t("react.suspense_list"), p = t("react.memo"), E = t("react.lazy"), R = t("react.block"), A = t("react.server.block"), h = t("react.fundamental"), y = t("react.debug_trace_mode"), S = t("react.legacy_hidden");
  }
  function _(e) {
    if (typeof e == "object" && e !== null) {
      var g = e.$$typeof;
      switch (g) {
        case s:
          switch (e = e.type, e) {
            case c:
            case l:
            case i:
            case u:
            case C:
              return e;
            default:
              switch (e = e && e.$$typeof, e) {
                case v:
                case n:
                case E:
                case p:
                case d:
                  return e;
                default:
                  return g;
              }
          }
        case f:
          return g;
      }
    }
  }
  var w = d, m = s, x = n, I = c, b = E, Y = p, L = f, O = l, D = i, k = u;
  return a.ContextConsumer = v, a.ContextProvider = w, a.Element = m, a.ForwardRef = x, a.Fragment = I, a.Lazy = b, a.Memo = Y, a.Portal = L, a.Profiler = O, a.StrictMode = D, a.Suspense = k, a.isAsyncMode = function() {
    return !1;
  }, a.isConcurrentMode = function() {
    return !1;
  }, a.isContextConsumer = function(e) {
    return _(e) === v;
  }, a.isContextProvider = function(e) {
    return _(e) === d;
  }, a.isElement = function(e) {
    return typeof e == "object" && e !== null && e.$$typeof === s;
  }, a.isForwardRef = function(e) {
    return _(e) === n;
  }, a.isFragment = function(e) {
    return _(e) === c;
  }, a.isLazy = function(e) {
    return _(e) === E;
  }, a.isMemo = function(e) {
    return _(e) === p;
  }, a.isPortal = function(e) {
    return _(e) === f;
  }, a.isProfiler = function(e) {
    return _(e) === l;
  }, a.isStrictMode = function(e) {
    return _(e) === i;
  }, a.isSuspense = function(e) {
    return _(e) === u;
  }, a.isValidElementType = function(e) {
    return typeof e == "string" || typeof e == "function" || e === c || e === l || e === y || e === i || e === u || e === C || e === S || typeof e == "object" && e !== null && (e.$$typeof === E || e.$$typeof === p || e.$$typeof === d || e.$$typeof === v || e.$$typeof === n || e.$$typeof === h || e.$$typeof === R || e[0] === A);
  }, a.typeOf = _, a;
}
var o = {};
/** @license React v17.0.2
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var W;
function Ce() {
  return W || (W = 1, process.env.NODE_ENV !== "production" && function() {
    var s = 60103, f = 60106, c = 60107, i = 60108, l = 60114, d = 60109, v = 60110, n = 60112, u = 60113, C = 60120, p = 60115, E = 60116, R = 60121, A = 60122, h = 60117, y = 60129, S = 60131;
    if (typeof Symbol == "function" && Symbol.for) {
      var t = Symbol.for;
      s = t("react.element"), f = t("react.portal"), c = t("react.fragment"), i = t("react.strict_mode"), l = t("react.profiler"), d = t("react.provider"), v = t("react.context"), n = t("react.forward_ref"), u = t("react.suspense"), C = t("react.suspense_list"), p = t("react.memo"), E = t("react.lazy"), R = t("react.block"), A = t("react.server.block"), h = t("react.fundamental"), t("react.scope"), t("react.opaque.id"), y = t("react.debug_trace_mode"), t("react.offscreen"), S = t("react.legacy_hidden");
    }
    var _ = !1;
    function w(r) {
      return !!(typeof r == "string" || typeof r == "function" || r === c || r === l || r === y || r === i || r === u || r === C || r === S || _ || typeof r == "object" && r !== null && (r.$$typeof === E || r.$$typeof === p || r.$$typeof === d || r.$$typeof === v || r.$$typeof === n || r.$$typeof === h || r.$$typeof === R || r[0] === A));
    }
    function m(r) {
      if (typeof r == "object" && r !== null) {
        var V = r.$$typeof;
        switch (V) {
          case s:
            var M = r.type;
            switch (M) {
              case c:
              case l:
              case i:
              case u:
              case C:
                return M;
              default:
                var q = M && M.$$typeof;
                switch (q) {
                  case v:
                  case n:
                  case E:
                  case p:
                  case d:
                    return q;
                  default:
                    return V;
                }
            }
          case f:
            return V;
        }
      }
    }
    var x = v, I = d, b = s, Y = n, L = c, O = E, D = p, k = f, e = l, g = i, J = u, K = !1, j = !1;
    function Q(r) {
      return K || (K = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    function ee(r) {
      return j || (j = !0, console.warn("The ReactIs.isConcurrentMode() alias has been deprecated, and will be removed in React 18+.")), !1;
    }
    function re(r) {
      return m(r) === v;
    }
    function te(r) {
      return m(r) === d;
    }
    function ne(r) {
      return typeof r == "object" && r !== null && r.$$typeof === s;
    }
    function ae(r) {
      return m(r) === n;
    }
    function oe(r) {
      return m(r) === c;
    }
    function ce(r) {
      return m(r) === E;
    }
    function se(r) {
      return m(r) === p;
    }
    function ie(r) {
      return m(r) === f;
    }
    function ue(r) {
      return m(r) === l;
    }
    function fe(r) {
      return m(r) === i;
    }
    function le(r) {
      return m(r) === u;
    }
    o.ContextConsumer = x, o.ContextProvider = I, o.Element = b, o.ForwardRef = Y, o.Fragment = L, o.Lazy = O, o.Memo = D, o.Portal = k, o.Profiler = e, o.StrictMode = g, o.Suspense = J, o.isAsyncMode = Q, o.isConcurrentMode = ee, o.isContextConsumer = re, o.isContextProvider = te, o.isElement = ne, o.isForwardRef = ae, o.isFragment = oe, o.isLazy = ce, o.isMemo = se, o.isPortal = ie, o.isProfiler = ue, o.isStrictMode = fe, o.isSuspense = le, o.isValidElementType = w, o.typeOf = m;
  }()), o;
}
(function(s) {
  process.env.NODE_ENV === "production" ? s.exports = me() : s.exports = Ce();
})(Z);
function Te(s, f) {
  let c = 0;
  function i(l) {
    F.Children.forEach(l, (d) => {
      Z.exports.isFragment(d) ? i(d.props.children) : (f(d, c), c += 1);
    });
  }
  i(s);
}
const ye = () => null, $ = "adm-tab-bar", $e = {
  safeArea: !1
}, Se = (s) => {
  var f;
  const c = pe($e, s);
  let i = null;
  const l = [];
  Te(c.children, (n, u) => {
    if (!F.isValidElement(n))
      return;
    const C = n.key;
    typeof C == "string" && (u === 0 && (i = C), l.push(n));
  });
  const [d, v] = ve({
    value: c.activeKey,
    defaultValue: (f = c.defaultActiveKey) !== null && f !== void 0 ? f : i,
    onChange: (n) => {
      var u;
      n !== null && ((u = c.onChange) === null || u === void 0 || u.call(c, n));
    }
  });
  return z(c, /* @__PURE__ */ B("div", {
    className: $,
    children: [/* @__PURE__ */ T("div", {
      className: `${$}-wrap`,
      children: l.map((n) => {
        const u = n.key === d;
        function C() {
          const p = n.props.icon && /* @__PURE__ */ T("div", {
            className: `${$}-item-icon`,
            children: typeof n.props.icon == "function" ? n.props.icon(u) : n.props.icon
          }), E = n.props.title && /* @__PURE__ */ T("div", {
            className: N(`${$}-item-title`, Boolean(p) && `${$}-item-title-with-icon`),
            children: typeof n.props.title == "function" ? n.props.title(u) : n.props.title
          });
          return p ? /* @__PURE__ */ B(H, {
            children: [/* @__PURE__ */ T(G, {
              content: n.props.badge,
              className: `${$}-icon-badge`,
              children: p
            }), E]
          }) : E ? /* @__PURE__ */ T(G, {
            content: n.props.badge,
            className: `${$}-title-badge`,
            children: E
          }) : null;
        }
        return z(n.props, /* @__PURE__ */ T("div", {
          onClick: () => {
            const {
              key: p
            } = n;
            p != null && v(p.toString());
          },
          className: N(`${$}-item`, {
            [`${$}-item-active`]: u
          }),
          children: C()
        }, n.key));
      })
    }), c.safeArea && /* @__PURE__ */ T(Ee, {
      position: "bottom"
    })]
  }));
};
export {
  Se as TabBar,
  ye as TabBarItem
};
