// lowcoder-sdk/dist/3ff7fd25.js
import { m as de, u as pe, d as fe, w as ge, a as me } from "./dbce84da.js";
import { useCallback as z, forwardRef as ye, useMemo as Q, useState as he, useEffect as ke } from "react";
import { u as we, b as ve, P as Se, r as xe } from "./483d0d71.js";
import { c as L, r as k, j as B } from "./8c706935.js";
import "react-dom";
var Z = { exports: {} };
(function(t, n) {
  (function(e, o) {
    t.exports = o();
  })(L, function() {
    var e = "day";
    return function(o, r, c) {
      var s = function(u) {
        return u.add(4 - u.isoWeekday(), e);
      }, i = r.prototype;
      i.isoWeekYear = function() {
        return s(this).year();
      }, i.isoWeek = function(u) {
        if (!this.$utils().u(u))
          return this.add(7 * (u - this.isoWeek()), e);
        var p, g, a, d, y = s(this), M = (p = this.isoWeekYear(), g = this.$u, a = (g ? c.utc : c)().year(p).startOf("year"), d = 4 - a.isoWeekday(), a.isoWeekday() > 4 && (d += 7), a.add(d, e));
        return y.diff(M, "week") + 1;
      }, i.isoWeekday = function(u) {
        return this.$utils().u(u) ? this.day() || 7 : this.day(this.day() % 7 ? u : u - 7);
      };
      var w = i.startOf;
      i.startOf = function(u, p) {
        var g = this.$utils(), a = !!g.u(p) || p;
        return g.p(u) === "isoweek" ? a ? this.date(this.date() - (this.isoWeekday() - 1)).startOf("day") : this.date(this.date() - 1 - (this.isoWeekday() - 1) + 7).endOf("day") : w.bind(this)(u, p);
      };
    };
  });
})(Z);
const ee = Z.exports;
var te = { exports: {} };
(function(t, n) {
  (function(e, o) {
    t.exports = o();
  })(L, function() {
    return function(e, o) {
      o.prototype.isoWeeksInYear = function() {
        var r = this.isLeapYear(), c = this.endOf("y").day();
        return c === 4 || r && c === 5 ? 53 : 52;
      };
    };
  });
})(te);
const ne = te.exports;
var oe = { exports: {} };
(function(t, n) {
  (function(e, o) {
    t.exports = o();
  })(L, function() {
    return function(e, o) {
      o.prototype.isLeapYear = function() {
        return this.$y % 4 == 0 && this.$y % 100 != 0 || this.$y % 400 == 0;
      };
    };
  });
})(oe);
const re = oe.exports, O = "TILL_NOW";
k.extend(ee);
k.extend(ne);
k.extend(re);
const I = {
  year: 0,
  month: 1,
  day: 2,
  hour: 3,
  minute: 4,
  second: 5
};
function We(t, n, e, o, r, c, s) {
  const i = [], w = n.getFullYear(), u = n.getMonth() + 1, p = n.getDate(), g = n.getHours(), a = n.getMinutes(), d = n.getSeconds(), y = e.getFullYear(), M = e.getMonth() + 1, P = e.getDate(), A = e.getHours(), F = e.getMinutes(), R = e.getSeconds(), x = I[o], N = parseInt(t[0]), j = k(G([t[0], t[1], "1"])), Y = parseInt(t[1]), C = parseInt(t[2]), D = parseInt(t[3]), v = parseInt(t[4]), f = N === w, H = N === y, _ = f && Y === u, S = H && Y === M, V = _ && C === p, q = S && C === P, J = V && D === g, K = q && D === A, ae = J && v === a, ce = K && v === F, T = (h, W, m) => {
    let l = [];
    for (let $ = h; $ <= W; $++)
      l.push($);
    const ue = t.slice(0, I[m]), E = c == null ? void 0 : c[m];
    return E && typeof E == "function" && (l = l.filter(($) => E($, {
      get date() {
        const le = [...ue, $.toString()];
        return G(le);
      }
    }))), l;
  };
  if (x >= I.year) {
    const m = T(w, y, "year");
    i.push(m.map((l) => ({
      label: r("year", l),
      value: l.toString()
    })));
  }
  if (x >= I.month) {
    const m = T(f ? u : 1, H ? M : 12, "month");
    i.push(m.map((l) => ({
      label: r("month", l),
      value: l.toString()
    })));
  }
  if (x >= I.day) {
    const h = _ ? p : 1, W = S ? P : j.daysInMonth(), m = T(h, W, "day");
    i.push(m.map((l) => ({
      label: r("day", l),
      value: l.toString()
    })));
  }
  if (x >= I.hour) {
    const m = T(V ? g : 0, q ? A : 23, "hour");
    i.push(m.map((l) => ({
      label: r("hour", l),
      value: l.toString()
    })));
  }
  if (x >= I.minute) {
    const m = T(J ? a : 0, K ? F : 59, "minute");
    i.push(m.map((l) => ({
      label: r("minute", l),
      value: l.toString()
    })));
  }
  if (x >= I.second) {
    const m = T(ae ? d : 0, ce ? R : 59, "second");
    i.push(m.map((l) => ({
      label: r("second", l),
      value: l.toString()
    })));
  }
  if (s && (i[0].push({
    label: r("now", null),
    value: O
  }), O === (t == null ? void 0 : t[0])))
    for (let h = 1; h < i.length; h += 1)
      i[h] = [];
  return i;
}
function Ie(t) {
  return t ? [t.getFullYear().toString(), (t.getMonth() + 1).toString(), t.getDate().toString(), t.getHours().toString(), t.getMinutes().toString(), t.getSeconds().toString()] : [];
}
function G(t) {
  var n, e, o, r, c, s;
  const i = (n = t[0]) !== null && n !== void 0 ? n : "1900", w = (e = t[1]) !== null && e !== void 0 ? e : "1", u = (o = t[2]) !== null && o !== void 0 ? o : "1", p = (r = t[3]) !== null && r !== void 0 ? r : "0", g = (c = t[4]) !== null && c !== void 0 ? c : "0", a = (s = t[5]) !== null && s !== void 0 ? s : "0";
  return new Date(parseInt(i), parseInt(w) - 1, parseInt(u), parseInt(p), parseInt(g), parseInt(a));
}
k.extend(ee);
k.extend(ne);
k.extend(re);
const b = {
  year: 0,
  week: 1,
  "week-day": 2
};
function Me(t, n, e, o, r, c) {
  const s = [], i = n.getFullYear(), w = e.getFullYear(), u = b[o], p = parseInt(t[0]), g = p === i, a = p === w, d = k(n), y = k(e), M = d.isoWeek(), P = y.isoWeek(), A = d.isoWeekday(), F = y.isoWeekday(), R = parseInt(t[1]), x = g && R === M, N = a && R === P, j = k(`${p}-01-01`).isoWeeksInYear(), Y = (C, D, v) => {
    let f = [];
    for (let S = C; S <= D; S++)
      f.push(S);
    const H = t.slice(0, b[v]), _ = c == null ? void 0 : c[v];
    return _ && typeof _ == "function" && (f = f.filter((S) => _(S, {
      get date() {
        const V = [...H, S.toString()];
        return se(V);
      }
    }))), f;
  };
  if (u >= b.year) {
    const v = Y(i, w, "year");
    s.push(v.map((f) => ({
      label: r("year", f),
      value: f.toString()
    })));
  }
  if (u >= b.week) {
    const v = Y(g ? M : 1, a ? P : j, "week");
    s.push(v.map((f) => ({
      label: r("week", f),
      value: f.toString()
    })));
  }
  if (u >= b["week-day"]) {
    const v = Y(x ? A : 1, N ? F : 7, "week-day");
    s.push(v.map((f) => ({
      label: r("week-day", f),
      value: f.toString()
    })));
  }
  return s;
}
function Ce(t) {
  if (!t)
    return [];
  const n = k(t);
  return [n.isoWeekYear().toString(), n.isoWeek().toString(), n.isoWeekday().toString()];
}
function se(t) {
  var n, e, o;
  const r = (n = t[0]) !== null && n !== void 0 ? n : "1900", c = (e = t[1]) !== null && e !== void 0 ? e : "1", s = (o = t[2]) !== null && o !== void 0 ? o : "1";
  return k().year(parseInt(r)).isoWeek(parseInt(c)).isoWeekday(parseInt(s)).hour(0).minute(0).second(0).toDate();
}
const De = {
  year: 1,
  month: 2,
  day: 3,
  hour: 4,
  minute: 5,
  second: 6
}, Ye = (t, n) => {
  if (n.includes("week"))
    return Ce(t);
  {
    const e = n;
    return Ie(t).slice(0, De[e]);
  }
}, U = (t, n) => {
  if ((t == null ? void 0 : t[0]) === O) {
    const e = new Date();
    return e.tillNow = !0, e;
  }
  return n.includes("week") ? se(t) : G(t);
}, _e = (t, n, e, o, r, c, s) => o.startsWith("week") ? Me(t, n, e, o, r, c) : We(t, n, e, o, r, c, s);
function Te(t) {
  const {
    locale: n
  } = we();
  return z((e, o) => {
    if (t)
      return t(e, o);
    switch (e) {
      case "minute":
      case "second":
      case "hour":
        return ("0" + o.toString()).slice(-2);
      case "now":
        return n.DatePicker.tillNow;
      default:
        return o.toString();
    }
  }, [t]);
}
const X = new Date().getFullYear(), $e = {
  min: new Date(new Date().setFullYear(X - 10)),
  max: new Date(new Date().setFullYear(X + 10)),
  precision: "day",
  defaultValue: null
}, ie = ye((t, n) => {
  const e = de($e, t), {
    renderLabel: o
  } = e, [r, c] = pe({
    value: e.value,
    defaultValue: e.defaultValue,
    onChange: (a) => {
      var d;
      a !== null && ((d = e.onConfirm) === null || d === void 0 || d.call(e, a));
    }
  }), s = Q(() => new Date(), []), i = Te(o), w = Q(() => {
    let a = r != null ? r : s;
    return a.tillNow ? [O] : (a = new Date(ve(a.getTime(), e.min.getTime(), e.max.getTime())), Ye(a, e.precision));
  }, [r, e.precision, e.min, e.max]), u = z((a) => {
    const d = U(a, e.precision);
    c(d, !0);
  }, [c, e.precision]), p = fe((a) => {
    var d;
    const y = U(a, e.precision);
    (d = e.onSelect) === null || d === void 0 || d.call(e, y);
  }), g = z((a) => _e(a, e.min, e.max, e.precision, i, e.filter, e.tillNow), [e.min, e.max, e.precision, i, e.tillNow]);
  return ge(e, /* @__PURE__ */ B(Se, {
    ref: n,
    columns: g,
    value: w,
    onCancel: e.onCancel,
    onClose: e.onClose,
    closeOnMaskClick: e.closeOnMaskClick,
    visible: e.visible,
    confirmText: e.confirmText,
    cancelText: e.cancelText,
    onConfirm: u,
    onSelect: p,
    getContainer: e.getContainer,
    loading: e.loading,
    loadingContent: e.loadingContent,
    afterShow: e.afterShow,
    afterClose: e.afterClose,
    onClick: e.onClick,
    title: e.title,
    stopPropagation: e.stopPropagation,
    mouseWheel: e.mouseWheel,
    destroyOnClose: e.destroyOnClose,
    forceRender: e.forceRender,
    children: (a, d) => {
      var y;
      return (y = e.children) === null || y === void 0 ? void 0 : y.call(e, r, d);
    }
  }));
});
function Pe(t) {
  return new Promise((n) => {
    const o = xe(/* @__PURE__ */ B(() => {
      const [r, c] = he(!1);
      return ke(() => {
        c(!0);
      }, []), /* @__PURE__ */ B(ie, {
        ...Object.assign({}, t, {
          visible: r,
          onConfirm: (s) => {
            var i;
            (i = t.onConfirm) === null || i === void 0 || i.call(t, s), n(s);
          },
          onClose: () => {
            var s;
            (s = t.onClose) === null || s === void 0 || s.call(t), c(!1), n(null);
          },
          afterClose: () => {
            var s;
            (s = t.afterClose) === null || s === void 0 || s.call(t), o();
          }
        })
      });
    }, {}));
  });
}
const Ne = me(ie, {
  prompt: Pe,
  DATE_NOW: O
});
export {
  Ne as default
};
