// lowcoder-sdk/dist/4907bc38.js
import { m as gt, u as yt, b as $t, w as vt, a as wt } from "./47d4822c.js";
import { useCallback as et, forwardRef as Mt, useMemo as ot, useState as St, useEffect as kt } from "react";
import { u as Dt, b as xt, P as Wt, r as Yt } from "./ac933d31.js";
import { b as K, g as X, j as nt } from "./8dfa988b.js";
import "react-dom";
var at = { exports: {} };
(function(r, l) {
  (function(t, c) {
    r.exports = c();
  })(K, function() {
    var t = 1e3, c = 6e4, p = 36e5, g = "millisecond", d = "second", h = "minute", _ = "hour", y = "day", Y = "week", S = "month", $ = "quarter", M = "year", C = "date", P = "Invalid Date", U = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, Z = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, B = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(s) {
      var n = ["th", "st", "nd", "rd"], e = s % 100;
      return "[" + s + (n[(e - 20) % 10] || n[e] || n[0]) + "]";
    } }, R = function(s, n, e) {
      var i = String(s);
      return !i || i.length >= n ? s : "" + Array(n + 1 - i.length).join(e) + s;
    }, F = { s: R, z: function(s) {
      var n = -s.utcOffset(), e = Math.abs(n), i = Math.floor(e / 60), o = e % 60;
      return (n <= 0 ? "+" : "-") + R(i, 2, "0") + ":" + R(o, 2, "0");
    }, m: function s(n, e) {
      if (n.date() < e.date())
        return -s(e, n);
      var i = 12 * (e.year() - n.year()) + (e.month() - n.month()), o = n.clone().add(i, S), a = e - o < 0, u = n.clone().add(i + (a ? -1 : 1), S);
      return +(-(i + (e - o) / (a ? o - u : u - o)) || 0);
    }, a: function(s) {
      return s < 0 ? Math.ceil(s) || 0 : Math.floor(s);
    }, p: function(s) {
      return { M: S, y: M, w: Y, d: y, D: C, h: _, m: h, s: d, ms: g, Q: $ }[s] || String(s || "").toLowerCase().replace(/s$/, "");
    }, u: function(s) {
      return s === void 0;
    } }, N = "en", A = {};
    A[N] = B;
    var j = function(s) {
      return s instanceof I;
    }, b = function s(n, e, i) {
      var o;
      if (!n)
        return N;
      if (typeof n == "string") {
        var a = n.toLowerCase();
        A[a] && (o = a), e && (A[a] = e, o = a);
        var u = n.split("-");
        if (!o && u.length > 1)
          return s(u[0]);
      } else {
        var w = n.name;
        A[w] = n, o = w;
      }
      return !i && o && (N = o), o || !i && N;
    }, D = function(s, n) {
      if (j(s))
        return s.clone();
      var e = typeof n == "object" ? n : {};
      return e.date = s, e.args = arguments, new I(e);
    }, f = F;
    f.l = b, f.i = j, f.w = function(s, n) {
      return D(s, { locale: n.$L, utc: n.$u, x: n.$x, $offset: n.$offset });
    };
    var I = function() {
      function s(e) {
        this.$L = b(e.locale, null, !0), this.parse(e);
      }
      var n = s.prototype;
      return n.parse = function(e) {
        this.$d = function(i) {
          var o = i.date, a = i.utc;
          if (o === null)
            return /* @__PURE__ */ new Date(NaN);
          if (f.u(o))
            return /* @__PURE__ */ new Date();
          if (o instanceof Date)
            return new Date(o);
          if (typeof o == "string" && !/Z$/i.test(o)) {
            var u = o.match(U);
            if (u) {
              var w = u[2] - 1 || 0, k = (u[7] || "0").substring(0, 3);
              return a ? new Date(Date.UTC(u[1], w, u[3] || 1, u[4] || 0, u[5] || 0, u[6] || 0, k)) : new Date(u[1], w, u[3] || 1, u[4] || 0, u[5] || 0, u[6] || 0, k);
            }
          }
          return new Date(o);
        }(e), this.$x = e.x || {}, this.init();
      }, n.init = function() {
        var e = this.$d;
        this.$y = e.getFullYear(), this.$M = e.getMonth(), this.$D = e.getDate(), this.$W = e.getDay(), this.$H = e.getHours(), this.$m = e.getMinutes(), this.$s = e.getSeconds(), this.$ms = e.getMilliseconds();
      }, n.$utils = function() {
        return f;
      }, n.isValid = function() {
        return this.$d.toString() !== P;
      }, n.isSame = function(e, i) {
        var o = D(e);
        return this.startOf(i) <= o && o <= this.endOf(i);
      }, n.isAfter = function(e, i) {
        return D(e) < this.startOf(i);
      }, n.isBefore = function(e, i) {
        return this.endOf(i) < D(e);
      }, n.$g = function(e, i, o) {
        return f.u(e) ? this[i] : this.set(o, e);
      }, n.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, n.valueOf = function() {
        return this.$d.getTime();
      }, n.startOf = function(e, i) {
        var o = this, a = !!f.u(i) || i, u = f.p(e), w = function(T, O) {
          var V = f.w(o.$u ? Date.UTC(o.$y, O, T) : new Date(o.$y, O, T), o);
          return a ? V : V.endOf(y);
        }, k = function(T, O) {
          return f.w(o.toDate()[T].apply(o.toDate("s"), (a ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(O)), o);
        }, m = this.$W, x = this.$M, W = this.$D, v = "set" + (this.$u ? "UTC" : "");
        switch (u) {
          case M:
            return a ? w(1, 0) : w(31, 11);
          case S:
            return a ? w(1, x) : w(0, x + 1);
          case Y:
            var J = this.$locale().weekStart || 0, L = (m < J ? m + 7 : m) - J;
            return w(a ? W - L : W + (6 - L), x);
          case y:
          case C:
            return k(v + "Hours", 0);
          case _:
            return k(v + "Minutes", 1);
          case h:
            return k(v + "Seconds", 2);
          case d:
            return k(v + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, n.endOf = function(e) {
        return this.startOf(e, !1);
      }, n.$set = function(e, i) {
        var o, a = f.p(e), u = "set" + (this.$u ? "UTC" : ""), w = (o = {}, o[y] = u + "Date", o[C] = u + "Date", o[S] = u + "Month", o[M] = u + "FullYear", o[_] = u + "Hours", o[h] = u + "Minutes", o[d] = u + "Seconds", o[g] = u + "Milliseconds", o)[a], k = a === y ? this.$D + (i - this.$W) : i;
        if (a === S || a === M) {
          var m = this.clone().set(C, 1);
          m.$d[w](k), m.init(), this.$d = m.set(C, Math.min(this.$D, m.daysInMonth())).$d;
        } else
          w && this.$d[w](k);
        return this.init(), this;
      }, n.set = function(e, i) {
        return this.clone().$set(e, i);
      }, n.get = function(e) {
        return this[f.p(e)]();
      }, n.add = function(e, i) {
        var o, a = this;
        e = Number(e);
        var u = f.p(i), w = function(x) {
          var W = D(a);
          return f.w(W.date(W.date() + Math.round(x * e)), a);
        };
        if (u === S)
          return this.set(S, this.$M + e);
        if (u === M)
          return this.set(M, this.$y + e);
        if (u === y)
          return w(1);
        if (u === Y)
          return w(7);
        var k = (o = {}, o[h] = c, o[_] = p, o[d] = t, o)[u] || 1, m = this.$d.getTime() + e * k;
        return f.w(m, this);
      }, n.subtract = function(e, i) {
        return this.add(-1 * e, i);
      }, n.format = function(e) {
        var i = this, o = this.$locale();
        if (!this.isValid())
          return o.invalidDate || P;
        var a = e || "YYYY-MM-DDTHH:mm:ssZ", u = f.z(this), w = this.$H, k = this.$m, m = this.$M, x = o.weekdays, W = o.months, v = function(O, V, tt, Q) {
          return O && (O[V] || O(i, a)) || tt[V].slice(0, Q);
        }, J = function(O) {
          return f.s(w % 12 || 12, O, "0");
        }, L = o.meridiem || function(O, V, tt) {
          var Q = O < 12 ? "AM" : "PM";
          return tt ? Q.toLowerCase() : Q;
        }, T = { YY: String(this.$y).slice(-2), YYYY: this.$y, M: m + 1, MM: f.s(m + 1, 2, "0"), MMM: v(o.monthsShort, m, W, 3), MMMM: v(W, m), D: this.$D, DD: f.s(this.$D, 2, "0"), d: String(this.$W), dd: v(o.weekdaysMin, this.$W, x, 2), ddd: v(o.weekdaysShort, this.$W, x, 3), dddd: x[this.$W], H: String(w), HH: f.s(w, 2, "0"), h: J(1), hh: J(2), a: L(w, k, !0), A: L(w, k, !1), m: String(k), mm: f.s(k, 2, "0"), s: String(this.$s), ss: f.s(this.$s, 2, "0"), SSS: f.s(this.$ms, 3, "0"), Z: u };
        return a.replace(Z, function(O, V) {
          return V || T[O] || u.replace(":", "");
        });
      }, n.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, n.diff = function(e, i, o) {
        var a, u = f.p(i), w = D(e), k = (w.utcOffset() - this.utcOffset()) * c, m = this - w, x = f.m(this, w);
        return x = (a = {}, a[M] = x / 12, a[S] = x, a[$] = x / 3, a[Y] = (m - k) / 6048e5, a[y] = (m - k) / 864e5, a[_] = m / p, a[h] = m / c, a[d] = m / t, a)[u] || m, o ? x : f.a(x);
      }, n.daysInMonth = function() {
        return this.endOf(S).$D;
      }, n.$locale = function() {
        return A[this.$L];
      }, n.locale = function(e, i) {
        if (!e)
          return this.$L;
        var o = this.clone(), a = b(e, i, !0);
        return a && (o.$L = a), o;
      }, n.clone = function() {
        return f.w(this.$d, this);
      }, n.toDate = function() {
        return new Date(this.valueOf());
      }, n.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, n.toISOString = function() {
        return this.$d.toISOString();
      }, n.toString = function() {
        return this.$d.toUTCString();
      }, s;
    }(), z = I.prototype;
    return D.prototype = z, [["$ms", g], ["$s", d], ["$m", h], ["$H", _], ["$W", y], ["$M", S], ["$y", M], ["$D", C]].forEach(function(s) {
      z[s[1]] = function(n) {
        return this.$g(n, s[0], s[1]);
      };
    }), D.extend = function(s, n) {
      return s.$i || (s(n, I, D), s.$i = !0), D;
    }, D.locale = b, D.isDayjs = j, D.unix = function(s) {
      return D(1e3 * s);
    }, D.en = A[N], D.Ls = A, D.p = {}, D;
  });
})(at);
var It = at.exports;
const H = /* @__PURE__ */ X(It);
var ut = { exports: {} };
(function(r, l) {
  (function(t, c) {
    r.exports = c();
  })(K, function() {
    var t = "day";
    return function(c, p, g) {
      var d = function(y) {
        return y.add(4 - y.isoWeekday(), t);
      }, h = p.prototype;
      h.isoWeekYear = function() {
        return d(this).year();
      }, h.isoWeek = function(y) {
        if (!this.$utils().u(y))
          return this.add(7 * (y - this.isoWeek()), t);
        var Y, S, $, M, C = d(this), P = (Y = this.isoWeekYear(), S = this.$u, $ = (S ? g.utc : g)().year(Y).startOf("year"), M = 4 - $.isoWeekday(), $.isoWeekday() > 4 && (M += 7), $.add(M, t));
        return C.diff(P, "week") + 1;
      }, h.isoWeekday = function(y) {
        return this.$utils().u(y) ? this.day() || 7 : this.day(this.day() % 7 ? y : y - 7);
      };
      var _ = h.startOf;
      h.startOf = function(y, Y) {
        var S = this.$utils(), $ = !!S.u(Y) || Y;
        return S.p(y) === "isoweek" ? $ ? this.date(this.date() - (this.isoWeekday() - 1)).startOf("day") : this.date(this.date() - 1 - (this.isoWeekday() - 1) + 7).endOf("day") : _.bind(this)(y, Y);
      };
    };
  });
})(ut);
var Ct = ut.exports;
const ct = /* @__PURE__ */ X(Ct);
var lt = { exports: {} };
(function(r, l) {
  (function(t, c) {
    r.exports = c();
  })(K, function() {
    return function(t, c) {
      c.prototype.isoWeeksInYear = function() {
        var p = this.isLeapYear(), g = this.endOf("y").day();
        return g === 4 || p && g === 5 ? 53 : 52;
      };
    };
  });
})(lt);
var _t = lt.exports;
const ft = /* @__PURE__ */ X(_t);
var dt = { exports: {} };
(function(r, l) {
  (function(t, c) {
    r.exports = c();
  })(K, function() {
    return function(t, c) {
      c.prototype.isLeapYear = function() {
        return this.$y % 4 == 0 && this.$y % 100 != 0 || this.$y % 400 == 0;
      };
    };
  });
})(dt);
var Ot = dt.exports;
const ht = /* @__PURE__ */ X(Ot), G = "TILL_NOW";
H.extend(ct);
H.extend(ft);
H.extend(ht);
const E = {
  year: 0,
  month: 1,
  day: 2,
  hour: 3,
  minute: 4,
  second: 5
};
function Tt(r, l, t, c, p, g, d) {
  const h = [], _ = l.getFullYear(), y = l.getMonth() + 1, Y = l.getDate(), S = l.getHours(), $ = l.getMinutes(), M = l.getSeconds(), C = t.getFullYear(), P = t.getMonth() + 1, U = t.getDate(), Z = t.getHours(), B = t.getMinutes(), R = t.getSeconds(), F = E[c], N = parseInt(r[0]), A = H(rt([r[0], r[1], "1"])), j = parseInt(r[1]), b = parseInt(r[2]), D = parseInt(r[3]), f = parseInt(r[4]), I = N === _, z = N === C, s = I && j === y, n = z && j === P, e = s && b === Y, i = n && b === U, o = e && D === S, a = i && D === Z, u = o && f === $, w = a && f === B, k = (m, x, W) => {
    let v = [];
    for (let T = m; T <= x; T++)
      v.push(T);
    const J = r.slice(0, E[W]), L = g == null ? void 0 : g[W];
    return L && typeof L == "function" && (v = v.filter((T) => L(T, {
      get date() {
        const O = [...J, T.toString()];
        return rt(O);
      }
    }))), v;
  };
  if (F >= E.year) {
    const W = k(_, C, "year");
    h.push(W.map((v) => ({
      label: p("year", v),
      value: v.toString()
    })));
  }
  if (F >= E.month) {
    const W = k(I ? y : 1, z ? P : 12, "month");
    h.push(W.map((v) => ({
      label: p("month", v),
      value: v.toString()
    })));
  }
  if (F >= E.day) {
    const m = s ? Y : 1, x = n ? U : A.daysInMonth(), W = k(m, x, "day");
    h.push(W.map((v) => ({
      label: p("day", v),
      value: v.toString()
    })));
  }
  if (F >= E.hour) {
    const W = k(e ? S : 0, i ? Z : 23, "hour");
    h.push(W.map((v) => ({
      label: p("hour", v),
      value: v.toString()
    })));
  }
  if (F >= E.minute) {
    const W = k(o ? $ : 0, a ? B : 59, "minute");
    h.push(W.map((v) => ({
      label: p("minute", v),
      value: v.toString()
    })));
  }
  if (F >= E.second) {
    const W = k(u ? M : 0, w ? R : 59, "second");
    h.push(W.map((v) => ({
      label: p("second", v),
      value: v.toString()
    })));
  }
  if (d && (h[0].push({
    label: p("now", null),
    value: G
  }), G === (r == null ? void 0 : r[0])))
    for (let m = 1; m < h.length; m += 1)
      h[m] = [];
  return h;
}
function bt(r) {
  return r ? [r.getFullYear().toString(), (r.getMonth() + 1).toString(), r.getDate().toString(), r.getHours().toString(), r.getMinutes().toString(), r.getSeconds().toString()] : [];
}
function rt(r) {
  var l, t, c, p, g, d;
  const h = (l = r[0]) !== null && l !== void 0 ? l : "1900", _ = (t = r[1]) !== null && t !== void 0 ? t : "1", y = (c = r[2]) !== null && c !== void 0 ? c : "1", Y = (p = r[3]) !== null && p !== void 0 ? p : "0", S = (g = r[4]) !== null && g !== void 0 ? g : "0", $ = (d = r[5]) !== null && d !== void 0 ? d : "0";
  return new Date(parseInt(h), parseInt(_) - 1, parseInt(y), parseInt(Y), parseInt(S), parseInt($));
}
H.extend(ct);
H.extend(ft);
H.extend(ht);
const q = {
  year: 0,
  week: 1,
  "week-day": 2
};
function Ht(r, l, t, c, p, g) {
  const d = [], h = l.getFullYear(), _ = t.getFullYear(), y = q[c], Y = parseInt(r[0]), S = Y === h, $ = Y === _, M = H(l), C = H(t), P = M.isoWeek(), U = C.isoWeek(), Z = M.isoWeekday(), B = C.isoWeekday(), R = parseInt(r[1]), F = S && R === P, N = $ && R === U, A = H(`${Y}-01-01`).isoWeeksInYear(), j = (b, D, f) => {
    let I = [];
    for (let n = b; n <= D; n++)
      I.push(n);
    const z = r.slice(0, q[f]), s = g == null ? void 0 : g[f];
    return s && typeof s == "function" && (I = I.filter((n) => s(n, {
      get date() {
        const e = [...z, n.toString()];
        return pt(e);
      }
    }))), I;
  };
  if (y >= q.year) {
    const f = j(h, _, "year");
    d.push(f.map((I) => ({
      label: p("year", I),
      value: I.toString()
    })));
  }
  if (y >= q.week) {
    const f = j(S ? P : 1, $ ? U : A, "week");
    d.push(f.map((I) => ({
      label: p("week", I),
      value: I.toString()
    })));
  }
  if (y >= q["week-day"]) {
    const f = j(F ? Z : 1, N ? B : 7, "week-day");
    d.push(f.map((I) => ({
      label: p("week-day", I),
      value: I.toString()
    })));
  }
  return d;
}
function At(r) {
  if (!r)
    return [];
  const l = H(r);
  return [l.isoWeekYear().toString(), l.isoWeek().toString(), l.isoWeekday().toString()];
}
function pt(r) {
  var l, t, c;
  const p = (l = r[0]) !== null && l !== void 0 ? l : "1900", g = (t = r[1]) !== null && t !== void 0 ? t : "1", d = (c = r[2]) !== null && c !== void 0 ? c : "1";
  return H().year(parseInt(p)).isoWeek(parseInt(g)).isoWeekday(parseInt(d)).hour(0).minute(0).second(0).toDate();
}
const Pt = {
  year: 1,
  month: 2,
  day: 3,
  hour: 4,
  minute: 5,
  second: 6
}, Ft = (r, l) => {
  if (l.includes("week"))
    return At(r);
  {
    const t = l;
    return bt(r).slice(0, Pt[t]);
  }
}, st = (r, l) => {
  if ((r == null ? void 0 : r[0]) === G) {
    const t = /* @__PURE__ */ new Date();
    return t.tillNow = !0, t;
  }
  return l.includes("week") ? pt(r) : rt(r);
}, Nt = (r, l, t, c, p, g, d) => c.startsWith("week") ? Ht(r, l, t, c, p, g) : Tt(r, l, t, c, p, g, d);
function jt(r) {
  const {
    locale: l
  } = Dt();
  return et((t, c) => {
    if (r)
      return r(t, c);
    switch (t) {
      case "minute":
      case "second":
      case "hour":
        return ("0" + c.toString()).slice(-2);
      case "now":
        return l.DatePicker.tillNow;
      default:
        return c.toString();
    }
  }, [r]);
}
const it = (/* @__PURE__ */ new Date()).getFullYear(), Lt = {
  min: new Date((/* @__PURE__ */ new Date()).setFullYear(it - 10)),
  max: new Date((/* @__PURE__ */ new Date()).setFullYear(it + 10)),
  precision: "day",
  defaultValue: null
}, mt = Mt((r, l) => {
  const t = gt(Lt, r), {
    renderLabel: c
  } = t, [p, g] = yt({
    value: t.value,
    defaultValue: t.defaultValue,
    onChange: ($) => {
      var M;
      $ !== null && ((M = t.onConfirm) === null || M === void 0 || M.call(t, $));
    }
  }), d = ot(() => /* @__PURE__ */ new Date(), []), h = jt(c), _ = ot(() => {
    let $ = p ?? d;
    return $.tillNow ? [G] : ($ = new Date(xt($.getTime(), t.min.getTime(), t.max.getTime())), Ft($, t.precision));
  }, [p, t.precision, t.min, t.max]), y = et(($) => {
    const M = st($, t.precision);
    g(M, !0);
  }, [g, t.precision]), Y = $t(($) => {
    var M;
    const C = st($, t.precision);
    (M = t.onSelect) === null || M === void 0 || M.call(t, C);
  }), S = et(($) => Nt($, t.min, t.max, t.precision, h, t.filter, t.tillNow), [t.min, t.max, t.precision, h, t.tillNow]);
  return vt(t, /* @__PURE__ */ nt(Wt, {
    ref: l,
    columns: S,
    value: _,
    onCancel: t.onCancel,
    onClose: t.onClose,
    closeOnMaskClick: t.closeOnMaskClick,
    visible: t.visible,
    confirmText: t.confirmText,
    cancelText: t.cancelText,
    onConfirm: y,
    onSelect: Y,
    getContainer: t.getContainer,
    loading: t.loading,
    loadingContent: t.loadingContent,
    afterShow: t.afterShow,
    afterClose: t.afterClose,
    onClick: t.onClick,
    title: t.title,
    stopPropagation: t.stopPropagation,
    mouseWheel: t.mouseWheel,
    destroyOnClose: t.destroyOnClose,
    forceRender: t.forceRender,
    children: ($, M) => {
      var C;
      return (C = t.children) === null || C === void 0 ? void 0 : C.call(t, p, M);
    }
  }));
});
function Rt(r) {
  return new Promise((l) => {
    const c = Yt(/* @__PURE__ */ nt(() => {
      const [p, g] = St(!1);
      return kt(() => {
        g(!0);
      }, []), /* @__PURE__ */ nt(mt, {
        ...Object.assign({}, r, {
          visible: p,
          onConfirm: (d) => {
            var h;
            (h = r.onConfirm) === null || h === void 0 || h.call(r, d), l(d);
          },
          onClose: () => {
            var d;
            (d = r.onClose) === null || d === void 0 || d.call(r), g(!1), l(null);
          },
          afterClose: () => {
            var d;
            (d = r.afterClose) === null || d === void 0 || d.call(r), c();
          }
        })
      });
    }, {}));
  });
}
const Zt = wt(mt, {
  prompt: Rt,
  DATE_NOW: G
});
export {
  Zt as default
};
