// lowcoder-sdk/dist/b3150ba7.js
import { c as y, P as M } from "./483d0d71.js";
import { s as C, e as T, a as g, j as a, T as s, t as u, S as w, C as k, h as f, m as b, n as I, o as d } from "./8c706935.js";
import "./dbce84da.js";
import "react";
import "react-dom";
y && document.addEventListener("touchstart", () => {
}, !0);
const S = (e = 1) => [...d.exports.range(0, 24, e).map((t) => {
  const n = (t < 10 ? "0" : "") + t;
  return {
    label: n,
    value: n
  };
})], $ = (e = 1) => [...d.exports.range(0, 12, e).map((t) => {
  const n = (t < 10 ? "0" : "") + t;
  return {
    label: n,
    value: n
  };
})], h = (e = 1) => d.exports.range(0, 60, e).map((t) => {
  const n = (t < 10 ? "0" : "") + t;
  return {
    label: n,
    value: n
  };
}), j = [{
  label: "AM",
  value: "am"
}, {
  label: "PM",
  value: "pm"
}], r = (e) => {
  const {
    disabledHours: t,
    disabledMinutes: n,
    disabledSeconds: x
  } = e.disabledTime();
  M.prompt({
    getContainer: () => document.querySelector(`#${k}`) || document.body,
    mouseWheel: !0,
    destroyOnClose: !0,
    closeOnMaskClick: !0,
    columns: (i) => {
      const l = f(i.join(":"), b);
      return [(e.use12Hours ? $ : S)(e.hourStep).filter(({
        label: m,
        value: o
      }) => !t().includes(Number(o) + (i[3] === "pm" ? 12 : 0))), h(e.minuteStep).filter(({
        label: m,
        value: o
      }) => !n(l.hour()).includes(Number(o))), h(e.secondStep).filter(({
        label: m,
        value: o
      }) => !x(l.hour(), l.minute()).includes(Number(o))), ...e.use12Hours ? [j] : []];
    },
    defaultValue: e.value ? e.value.format(e.use12Hours ? I : s).split(":") : void 0,
    onConfirm: (i) => {
      const l = f(i.join(":"), b);
      e.onChange(l);
    },
    onClose: e.onBlur
  }), e.onFocus();
}, v = C.div`
  height: 32px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 4px 8px;
  background-color: #ffffff;
  cursor: pointer;
  border-radius: 4px;
  border: 1px solid #d7d9e0;

  ${(e) => e.$style && T(e.$style)}
`, c = C.div`
  overflow: hidden;
  white-space: nowrap;
  flex-grow: 1;
  display: flex;
  justify-content: center;
`, V = (e) => /* @__PURE__ */ g(v, {
  ref: e.viewRef,
  $style: e.$style,
  onClick: () => {
    r(e);
  },
  children: [/* @__PURE__ */ a(c, {
    children: e.value ? e.value.format(e.format || s) : u("time.placeholder")
  }), e.suffixIcon]
}), E = (e) => /* @__PURE__ */ g(v, {
  ref: e.viewRef,
  $style: e.$style,
  children: [/* @__PURE__ */ a(c, {
    onClick: () => r({
      ...e,
      value: e.start,
      onChange: (t) => e.onChange(t, e.end)
    }),
    children: e.start ? e.start.format(e.format || s) : u("time.startTime")
  }), /* @__PURE__ */ a(w, {}), /* @__PURE__ */ a(c, {
    onClick: () => r({
      ...e,
      value: e.end,
      onChange: (t) => e.onChange(e.start, t)
    }),
    children: e.end ? e.end.format(e.format || s) : u("time.endTime")
  }), e.suffixIcon]
});
export {
  V as TimeMobileUIView,
  E as TimeRangeMobileUIView
};
